qutebrowser/mainwindow/tabwidget.py

```diff
@@ -457,12 +450,35 @@ def minimumTabSizeHint(self, index):
             padding_h += self.style().pixelMetric(
                 PixelMetrics.icon_padding, None, self)
         height = self.fontMetrics().height() + padding_v
-        width = (self.fontMetrics().width('\u2026') + icon_size.width() +
+        width = (self.fontMetrics().width(text) + icon_size.width() +
                  padding_h + config.val.tabs.width.indicator)
         return QSize(width, height)
 
-    def tabSizeHint(self, index):
-        """Override tabSizeHint so all tabs are the same size.
+    def _tabTotalWidthPinned(self):
+        """Get the current total width of pinned tabs.
+
+        This width is calculated assuming no shortening due to elipsis."""
+        return sum(
+            # Get the width (int) from minimum size
+            map(lambda tab: tab.width(),
+                # Get the minimum size of tabs, no elipsis
+                map(functools.partial(self.minimumTabSizeHint, elipsis=False),
+                    # Get only pinned tabs (from indexes)
+                    filter(self._tabPinned, range(self.count())))))
+
+    def _pinnedCount(self) -> int:
+        """Get the number of pinned tabs."""
+        return len(list(filter(self._tabPinned, range(self.count()))))
```

Comment:
`sum(self._tab_pinned(idx) for idx in range(self.count()))`

(ab)using the fact that a boolean is also an int 😆